os/include/tinyara/binfmt/compression/compress_read.h

```diff
@@ -32,12 +32,18 @@
  * Public Types
  ****************************************************************************/
 
-/* Struct for buffers to be used for read/decompression */
-struct s_buffer {
-	unsigned char *read_buffer;
-	unsigned char *out_buffer;
+/* Struct for output buffers to cache uncompressed blocks */
+struct block_cache {
+	unsigned char *out_buffer;			/* Buffer that is going to hold uncompressed data */
+	int block_number;			/* Block number in compressed file for the cached block */
+	unsigned int no_requests_for_block;	/* Number of requests for current block that is cached */
+	unsigned int index_block_cache;		/* Index of block cache in the array */
+	struct block_cache *next;
+	struct block_cache *prev;
 };
 
+typedef struct block_cache block_cache;
```

Comment:
I think this will cause a compilation warning.
Let's use our rules.
```
struct block_cache_s {
...
};
typedef struct block_cache_s block_cache_t;
```

Reply:
done